src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -1,56 +1,50 @@
 package com.stanimiriliev.chaptertwo.homogeneoustree;
 
 /**
- * Binary tree
- * Created by Stanimir Iliev on 18.6.2017 y..
+ * Binary Ordered Tree, Homogeneous Tree
+ * @author  Stanimir
+ * @since 18.6.2017
  */
 public class Tree {
```

Comment:
the method is defined in the class Tree not in the class Node, previously left and right was Tree too.
maybe now you should have another method
contains(node, element) and you will pass use it for the recursion